highlight_view.py

```diff
@@ -0,0 +1,213 @@
+    return highlight_regions, gutter_regions
+
+
+def get_base_error_style(linter, code, error_type, **kwargs):
+    store = style_stores.get_linter_style_store(linter)
+    return store.get_style(code, error_type)
+
+
+def prepare_gutter_data(view, errors):
+    errors_by_line = defaultdict(list)
+    for error in errors:
+        priority = highlight_store.get(error['style']).get('priority', 0)
+        errors_by_line[error['line']].append(ChainMap({'priority': int(priority)}, error))
+
+    # Since, there can only be one gutter mark per line, we have to select
+    # one 'winning' error from all the errors on that line
+    error_per_line = {}
+    for line, errors in errors_by_line.items():
+        # We're lucky here that 'error' comes before 'warning'
+        head = sorted(errors, key=lambda e: (e['error_type'], -e['priority']))[0]
+        error_per_line[line] = head
+
+    # Compute the icon and scope for the gutter mark from the error.
+    # Drop lines for which we don't get a value or for which the user
+    # specified 'none'
+    by_id = defaultdict(list)
+    for line, error in error_per_line.items():
+
+        if not highlight_store.has_style(error['style']):  # really?
+            continue
+
+        icon = highlight_store.get_val('icon', error['style'], error['error_type'])
+        if not icon or icon == 'none':
+            continue
+
+        if style_stores.GUTTER_ICONS.get('colorize', True) or icon in INBUILT_ICONS:
+            scope = highlight_store.get_val('scope', error['style'], error['error_type'])
+        else:
+            scope = " "  # set scope to non-existent one
+
+        pos = get_line_start(view, line)
+        region = sublime.Region(pos, pos)
+
+        # We group towards the optimal sublime API usage:
+        #   view.add_regions(uuid(), [region], scope, icon)
+        id = (scope, icon)
+        by_id[id].append(region)
+
+    # Exchange the `id` with a regular region_id which is a unique string, so
+    # uuid() would be candidate here, that can be reused for efficient updates.
+    by_region_id = {}
+    for (scope, icon), regions in by_id.items():
+        region_id = 'SL.Gutter.{}.{}'.format(scope, icon)
+        by_region_id[region_id] = (scope, icon, regions)
+
+    return by_region_id
+
+
+def prepare_highlights_data(view, errors):
+    # We can only one highlight per exact position, so we first group per
+    # position.
+    by_position = defaultdict(list)
+    for error in errors:
+        line_start = get_line_start(view, error['line'])
+        region = sublime.Region(line_start + error['start'], line_start + error['end'])
+        by_position[(region.a, region.b)].append(ChainMap({'region': region}, error))
+
+    by_id = defaultdict(list)
+    for pos, errors in by_position.items():
+        # If we have multiple 'problems' here, 'error' takes precedence over
+        # 'warning'. We're lucky again that 'error' copmes before 'warning'
```

Comment:
Typo: copmes -> comes

Glad to be able to contribute something of value to this PR 💪🏻